style/testing/avoid_let_spec.rb

```diff
@@ -0,0 +1,38 @@
+describe SomeClass do
+  describe "#let" do
+    let(:user) { User.new(name: "John") }
+
+    context "avoid using" do
+      it "uses lets" do
+        # BAD
+        user.name = "Jane"
+        result = SomeClass.new(user).run
+
+        expect(result).to be_present
+      end
+
+      it "does not use lets" do
+        # GOOD
+        user = build_user(name: "Jane")
+
+        result = SomeClass.new(user).run
+
+        expect(result).to be_present
+      end
+    end
+
+    context "don't re-implement" do
+      it "uses a re-implementation of let" do
+        some_variable
+
+        expect(true).to be_true
+      end
+
+      def some_variable
+        # BAD -- just use let, or better extract a function that allows passing
+        # the thing that is different accross the examples.
```

Comment:
@gylaz instead of explaining this in a comment, can you add an example of the function you mention?

I'm not entirely sure what you mean by `a function that allows passing the thing that is different accross the examples`, so I feel an example would help here.